ipfs-4-to-5/migration/migration.go

```diff
@@ -0,0 +1,224 @@
+package mg3
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	migrate "github.com/ipfs/fs-repo-migrations/go-migrate"
+	lock "github.com/ipfs/fs-repo-migrations/ipfs-1-to-2/repolock"
+	mfsr "github.com/ipfs/fs-repo-migrations/mfsr"
+	log "github.com/ipfs/fs-repo-migrations/stump"
+
+	flatfs "github.com/ipfs/fs-repo-migrations/ipfs-4-to-5/go-ds-flatfs"
+)
+
+type Migration struct{}
+
+func (m Migration) Versions() string {
+	return "4-to-5"
+}
+
+func (m Migration) Reversible() bool {
+	return true
+}
+
+func revertStep2(ffspath string) error {
+	if err := flatfs.DowngradeV1toV0(ffspath); err != nil {
+		return fmt.Errorf("reverting flatfsv1 upgrade: %s", err)
+	}
+	return nil
+}
+
+func (m Migration) Apply(opts migrate.Options) error {
+	log.Verbose = opts.Verbose
+	log.Log("applying %s repo migration", m.Versions())
+
+	log.VLog("locking repo at %q", opts.Path)
+	lk, err := lock.Lock2(opts.Path)
+	if err != nil {
+		return err
+	}
+	defer lk.Close()
+
+	repo := mfsr.RepoPath(opts.Path)
+
+	log.VLog("  - verifying version is '4'")
+	if err := repo.CheckVersion("4"); err != nil {
+		return err
+	}
+
+	basepath := filepath.Join(opts.Path, "blocks")
+	ffspath := filepath.Join(opts.Path, "blocks-v4")
+	if err := os.Rename(basepath, ffspath); err != nil {
+		return err
+	}
+
+	revert1 := func(e error) error {
+		err := os.Rename(ffspath, basepath)
+		if err != nil {
+			log.Error(err)
+		}
+		return e
+	}
+
+	log.Log("> Upgrading datastore format to have sharding specification file")
```

Comment:
Is there any special reason some of the log entries start with `>`?  Does the `>` do something special?